domains/client/domain-operator/src/fraud_proof.rs

```diff
@@ -518,10 +518,14 @@ where
         mismatch_type: BundleMismatchType,
         bundle_index: u32,
         bad_receipt_hash: Block::Hash,
-        // Whether allow generate invalid proof against valid ER,
-        // only used in test
-        allow_invalid_proof: bool,
+        // Whether allow generating an invalid proof against a valid ER,
+        // only used in tests, ignored in production
+        mut allow_invalid_proof: bool,
```

Comment:
unfortunate that rust does not support conditional compilation for individual functional args

Reply:
The language does, but it isn't supported well by other tooling. So you end up with weird LSP or clippy errors sometimes.

What do you think of 332eef97d, which uses a wrapper function to drop the argument in production code?

Reply:
okay that works too!